jdk/src/share/classes/sun/security/ec/ECPrivateKeyImpl.java

```diff
@@ -213,4 +223,43 @@ protected void parseKeyBits() throws InvalidKeyException {
             throw new InvalidKeyException("Invalid EC private key", e);
         }
     }
+
+    /**
+     * Returns the native EC public key context pointer.
+     * @return the native EC public key context pointer or -1 on error
+     */
+    long getNativePtr() {
+        if (nativeECKey == 0x0) {
+            synchronized (this) {
+                if (nativeECKey == 0x0) {
+                    ECParameterSpec parameters = this.getParams();
+                    ECPoint generator = parameters.getGenerator();
+                    EllipticCurve curve = parameters.getCurve();
+                    ECField field = curve.getField();
+                    byte[] a = curve.getA().toByteArray();
+                    byte[] b = curve.getB().toByteArray();
+                    byte[] gx = generator.getAffineX().toByteArray();
+                    byte[] gy = generator.getAffineY().toByteArray();
+                    byte[] n = parameters.getOrder().toByteArray();
+                    byte[] h = BigInteger.valueOf(parameters.getCofactor()).toByteArray();
+                    byte[] p = new byte[0];
+                    if (field instanceof ECFieldFp) {
+                        p = ((ECFieldFp)field).getP().toByteArray();
+                        nativeECKey = nativeCrypto.ECEncodeGFp(a, a.length, b, b.length, p, p.length, gx, gx.length, gy, gy.length, n, n.length, h, h.length);
+                    } else if (field instanceof ECFieldF2m) {
+                        p = ((ECFieldF2m)field).getReductionPolynomial().toByteArray();
+                        nativeECKey = nativeCrypto.ECEncodeGF2m(a, a.length, b, b.length, p, p.length, gx, gx.length, gy, gy.length, n, n.length, h, h.length);
+                    }
+                    if (nativeECKey > 0)  {
```

Comment:
There should be an `else` block here which sets `nativeECKey = -1;` to match the javadoc and the test should be `nativeECKey != -1`. Likewise in `ECPublicKeyImpl`.